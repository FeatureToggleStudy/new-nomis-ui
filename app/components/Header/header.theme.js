import styled from 'styled-components';
// import colors from 'theme/colors';
import fonts from 'theme/fonts';

const height = '60px';

export const Base = styled.div`
  ${fonts.misc}
  display: flex;
  flex-direction: row;
  height: ${height};
  align-items: center;
  justify-content: space-between;
  color: white;
`;

export const Logo = styled.div`
  width: 40px;
  margin-left: 15px;
  padding-bottom: 6px;
`;

export const LogoText = styled.div`
  color: white;
  font-size: 26px;
  padding: 0 15px;
  font-weight: bold;
`;

export const Title = styled.div`
  flex-grow: 1;
  color: white;
  font-size: 26px;
  border-left: solid white 1px;
  padding-left: 15px;
  font-weight: bold;
`;

export const SearchButton = styled.div`
  margin: 0;
  padding: 0 20px;
`;

export const SearchIcon = styled.div`
  margin: 0;
  padding: 0 20px;
`;

export const SearchText = styled.div`
  margin: 0;
  padding: 0 20px;
`;

export const UserMenu = styled.div`
  margin: 0;
  padding: 0 20px;
`;
//
// const LiRef = () => `
//   display: inline-flex;
//   margin: 0 10px;
//   align-items: center;
//   height: ${height};
//   cursor: pointer;
// `;
//
// export const Li = styled.li`
//   ${LiRef}
// `;
//
// export const Hover = styled.li`
//   ${LiRef}
//   position: relative;
//   padding: 0 10px;
//   height: ${height};
//   background: transparent;
//   transition: background 400ms ease;
//   &:hover {
//     background: lighten($ig-blue, 10%);
//   }
// `;
