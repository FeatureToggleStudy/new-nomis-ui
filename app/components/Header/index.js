import React from 'react';
import PropTypes from 'prop-types';

import DesktopWrapper from 'components/CommonTheme/DesktopWrapper';
import colours from 'theme/colours';
import SVGLOGO from './svgLogo';

import {
  Base,
  Logo,
  LogoText,
  Title,
  // SearchButton,
  // SearchIcon,
  // SearchText,
  // UserMenu,
} from './header.theme';


function Header({ menuComponent }) {
  return (
    <DesktopWrapper background={colours.headerColour}>
      <Base>
        <Logo><SVGLOGO /></Logo>
        <LogoText>HMPPS</LogoText>
        <Title>Prison-NOMIS</Title>
        {/* <SearchButton>
          <SearchIcon />
          <SearchText></SearchText>
        </SearchButton>
        <UserMenu /> */}
        {menuComponent}
      </Base>
    </DesktopWrapper>
  );
}

Header.propTypes = {
  menuComponent: PropTypes.node,
  // LeftIcon: PropTypes.element,
  // Title: PropTypes.element,
  // RightIcon: PropTypes.element,
};

Header.defaultProps = {
  menuComponent: null,
};

export default Header;
