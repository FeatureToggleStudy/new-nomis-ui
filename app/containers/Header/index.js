import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

// eslint-disable-next-line import/no-unresolved
import HeaderComponent from 'header'
import { setMenuOpen, showTerms, navigateTo } from '../../globalReducers/app'
import { switchCaseLoad } from '../EliteApiLoader/actions'

const HeaderContainer = ({
  user,
  menuOpen,
  switchCaseLoad: switchCL,
  setMenuOpen: openMenu,
  showTermsAndConditions,
  navigateTo: toRoute,
}) => (
  <HeaderComponent
    switchCaseLoad={switchCL}
    user={user}
    menuOpen={menuOpen}
    setMenuOpen={openMenu}
    showTermsAndConditions={showTermsAndConditions}
    navigateTo={toRoute}
  />
)

HeaderContainer.propTypes = {
  showTermsAndConditions: PropTypes.func.isRequired,
  navigateTo: PropTypes.func.isRequired,
  menuOpen: PropTypes.bool,
  setMenuOpen: PropTypes.func,
  switchCaseLoad: PropTypes.func.isRequired,
  user: PropTypes.shape({ isKeyWorker: PropTypes.bool }),
}

HeaderContainer.defaultProps = {
  menuOpen: false,
  setMenuOpen: () => {},
  user: undefined,
}

const mapStateToProps = state => {
  const caseLoadOptions = state.getIn(['eliteApiLoader', 'User', 'CaseLoads', 'Data'])
  const user = state.getIn(['authentication', 'user'])

  return {
    menuOpen: state.getIn(['app', 'mobileMenuOpen']),
    user: {
      ...user,
      caseLoadOptions: (caseLoadOptions && caseLoadOptions.toJS()) || [],
    },
  }
}

const mapDispatchToProps = {
  setMenuOpen,
  switchCaseLoad,
  navigateTo,
  showTermsAndConditions: showTerms,
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HeaderContainer)
