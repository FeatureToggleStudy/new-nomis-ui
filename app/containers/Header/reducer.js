/*
 *
 * Header reducer
 *
 * Controls what is shown in the header.
 *
 */

import { fromJS } from 'immutable';
import {

} from './constants';

const initialState = fromJS({
  logo: true,
  logoText: 'HMPPS',
  title: 'Prison-NOMIS',
  search: true,
  userMenu: true,
});

function headerReducer(state = initialState, action) {
  switch (action.type) {
    // case LOGIN_SUCCESS: {
    //   const username = state.get('usernameInput');
    //   return state
    //     .set('user', { username, ...action.userObj })
    //     .set('loggedIn', true)
    //     .set('lastLogin', Date().toString())
    //     .set('passwordInput', '')
    //     .set('usernameInput', '');
    // }
    // case LOGOUT_SUCCESS: {
    //   return state
    //     .set('user', null)
    //     .set('loggedIn', false);
    // }
    // case CHANGE_USERNAME_INPUT: {
    //   return state
    //     .set('usernameInput', action.username);
    // }
    // case CHANGE_PASSWORD_INPUT: {
    //   return state
    //     .set('passwordInput', action.password);
    // }
    default: {
      return state;
    }
  }
}

export default headerReducer;
