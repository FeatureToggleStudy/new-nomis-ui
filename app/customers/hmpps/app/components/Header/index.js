  import React, { Component } from 'react';
  import PropTypes from 'prop-types';

  import Dropdown from 'components/Dropdown';
  import MenuToggle from 'components/MenuToggle';
  import ProductGlobals from 'product-globals';

  import {
  DesktopOnly,
  MobileOnly,
} from 'components/CommonTheme';

  import {
  PageHeader,
  LeftContent,
  RightContent,
  Logo,
  LogoText,
  Title,
  ToggleWrapper,
} from './header.theme';

  class Header extends Component {

    constructor(props) {
      super(props);

      this.openMenu = this.openMenu.bind(this);
      this.closeMenu = this.closeMenu.bind(this);
    }

    openMenu() {
      this.props.setMobileMenuOpen(true);
    }

    closeMenu() {
      this.props.setMobileMenuOpen(false);
    }

    render() {
      const { user, mobileMenuOpen, switchCaseLoad } = this.props;
      return (
      <PageHeader>
        <div className="header-content">
          <LeftContent>
            <Logo className="header-image">
            </Logo>
            <LogoText to="/">HMPPS</LogoText>
            <Title>{ProductGlobals.serviceName}</Title>
          </LeftContent>
          <RightContent>
            <DesktopOnly>
              {user && <Dropdown switchCaseLoad={switchCaseLoad} user={user} /> }
            </DesktopOnly>
            <MobileOnly>
              { user &&
                <ToggleWrapper>
                  <MenuToggle toggleState={mobileMenuOpen} onToggle={mobileMenuOpen ? this.closeMenu : this.openMenu} />
                </ToggleWrapper>
              }
            </MobileOnly>
          </RightContent>
        </div>
      </PageHeader>
      );
    }
}

  Header.propTypes = {
    user: PropTypes.object,
    mobileMenuOpen: PropTypes.bool,
    setMobileMenuOpen: PropTypes.func,
    switchCaseLoad: PropTypes.func.isRequired,
  };

  Header.contextTypes = {
    router: PropTypes.object.isRequired,
  };

  Header.defaultProps = {
    user: undefined,
    options: {
      assignments: 12,
      facilities: ['Sheffield', 'Cloverfield'],
    },
    mobileMenuOpen: false,
    setMobileMenuOpen: () => {},
  };

  export default Header;
