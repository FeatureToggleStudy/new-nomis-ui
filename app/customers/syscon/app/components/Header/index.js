import React from 'react';
import Dropdown from 'components/Dropdown';
import MenuToggle from 'components/MenuToggle';

import './index.scss';

export default ({ user, switchCaseLoad, mobileMenuOpen, setMobileMenuOpen }) => (
    <div className="banner">

      <div className="header-content">

           <div className="brand-header">

             <span className="section" to="/">
               <img src="/img/Syscon_logo.svg"></img>
             </span>

             <span className="section">
               <h1>
                 SYSCON
               </h1>
             </span>
             <span className="section desktop-only">
               <span className="divider"></span>
             </span>

             <span className="section desktop-only">
               <h1>
                 Prison Manager
               </h1>
             </span>
           </div>


          <div className="desktop-menu">
            {user && <Dropdown switchCaseLoad={switchCaseLoad} user={user} /> }
          </div>

          <div className="mobile-menu">
            {user &&
              <MenuToggle
                toggleState={mobileMenuOpen}
                onToggle={mobileMenuOpen ? () => setMobileMenuOpen(false) : () => setMobileMenuOpen(true)}
              />
            }

          </div>
      </div>

  </div>)
