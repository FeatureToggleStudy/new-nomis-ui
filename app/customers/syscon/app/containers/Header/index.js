import React from 'react';
import Dropdown from 'components/Dropdown';
import './index.scss';

export default ({ user, switchCaseLoad }) => (
  <div className="banner">

     <div className="logo">

       <img src="/img/Syscon_logo.svg"></img>

       <h1 className="company-name">
         SYSCON
       </h1>

       <span className="divider"></span>

       <h1 className="service-name">
         Prison Manager
       </h1>
       <div className="desktop-menu">
         {user && <Dropdown switchCaseLoad={switchCaseLoad} user={user} /> }
       </div>

       <div className="mobile-menu">
       </div>

     </div>

</div>)
