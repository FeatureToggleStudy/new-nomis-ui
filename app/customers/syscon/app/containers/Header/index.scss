
.banner {
  box-sizing: border-box;
  width: 100%;
  margin: 0;
  padding: 0;
  background: rgb(0, 39, 72);
  color: white;
  z-index: 999;
  height: 60px;

  @include media-breakpoint-down(sm){
    height: 55px!important;
  }

  .logo {
    display: flex;
    flex-direction: row;
    margin-bottom: 22rem;

    img{
      padding-left: 40px;
      padding-right: 15px;
      margin-bottom: .5rem;
    }

    .company-name {
      font-size: 24px;
      @include media-breakpoint-down(sm){
        font-size: 19px;
      }
      font-weight: bold;
      margin-bottom: 1rem;
    }

    .service-name {
      font-size: 24px;
      @include media-breakpoint-down(sm){
        font-size: 19px;
      }
      font-weight: lighter;
      margin-bottom: .5rem;
    }

    .divider {
      border-right: solid 2px white;
      margin: 15px 15px 15px 15px;
    }

    .desktop-menu {
      margin-left: auto;
      height: 60px;
    }

    @include media-breakpoint-down(sm){
      .desktop-menu {
        display: none;
      }
    }

  }
}