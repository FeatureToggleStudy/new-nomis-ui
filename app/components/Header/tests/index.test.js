import React from 'react';
import { shallow } from 'enzyme';

import LanguageSelect from 'containers/LanguageSelect';
import Header from '../index';
import {
  Base,
  Ul,
  Li,
  Hover,
} from '../header.theme';

const renderComponent = (props) => shallow(<Header {...props} />);

describe('Component: <Header>', () => {
  it('should match its snapshot with no props', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.getNodes()).toMatchSnapshot();
  });

  it('should match its snapshot with Name prop', () => {
    const renderedComponent = renderComponent({ name: 'Sophie' });
    expect(renderedComponent.getNodes()).toMatchSnapshot();
  });

  it('should have a base style with children', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.find(Base).length).toEqual(1);
  });

  it('should have at least one ul', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.find(Ul).length).toBeGreaterThanOrEqual(1);
  });

  it('should render navigation links', () => {
    const links = [
      { id: 'navOne', text: 'one', link: '' },
      { id: 'navTwo', text: 'two', link: '' },
      { id: 'navThree', text: 'three', link: '' },
    ];
    const renderedComponent = renderComponent({ links });
    expect(renderedComponent.find(Hover).length).toBeGreaterThanOrEqual(1);
  });

  it('should not render the Users name if no props are passed', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent.find(Li).length).toEqual(1);
  });

  it('should show the language select options', () => {
    const renderedComponent = renderComponent();
    expect(renderedComponent
      .find(Ul).at(1)
      .find(Li).last()
      .contains(<LanguageSelect />)
    ).toBe(true);
  });
});
