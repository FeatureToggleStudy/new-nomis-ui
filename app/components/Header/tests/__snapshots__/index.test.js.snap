exports[`Component: <Header> should match its snapshot with Name prop 1`] = `
Array [
  <styled.div>
    <div>
        <styled.ul />
    </div>
    <div>
        <styled.ul>
            <styled.li>
                <Connect(LocaleToggle) />
            </styled.li>
        </styled.ul>
    </div>
</styled.div>,
]
`;

exports[`Component: <Header> should match its snapshot with no props 1`] = `
Array [
  <styled.div>
    <div>
        <styled.ul />
    </div>
    <div>
        <styled.ul>
            <styled.li>
                <Connect(LocaleToggle) />
            </styled.li>
        </styled.ul>
    </div>
</styled.div>,
]
`;
